products/zoningtaxlots/models/product/qa/qa_mismatch.sql

```diff
@@ -0,0 +1,100 @@
+{{ config(
+    materialized = 'table'
+) }}
+
+WITH new_version AS (
+    SELECT * FROM {{ ref('int__zoningtaxlots') }}
+),
+
+prev_version AS (
+    SELECT * FROM {{ source('recipe_sources', 'dcp_zoningtaxlots') }}
+),
+
+mismatch AS (
+    SELECT
+        count(*) AS total,
+        sum(
+            (a.borough_code != b.borough_code OR (a.borough_code IS NULL AND b.borough_code IS NOT NULL))::integer
```

Comment:
sqlfluff is doing it's best, but what might make a lot of this cleaner is some macros.

There's some similar stuff in queries below in qa_bbldiffs. It seems that we could use a dbt macro that takes in
- left table
- right table
- left table column
- right table column (default to left table column)

which when called `{{ not_equal_or_null(a, b, borough_code) }}` would give something like the clause above. Or maybe additionally have a boolean flag argument - down in bbldiffs, we want the clause above with an additional `OR`